### PR TITLE
Feature/kak/strta#161

### LINKED ISSUE #161: Use STRTA
Scope expansion of #151 in #160.

Implement scripts to match cross-project standards as described [here](https://github.com/azavea/architecture/blob/master/doc/arch/adr-0000-scripts-to-rule-them-all.md).

### PR DESCRIPTION
**Pull request by flibbertigibbet**
## Overview

Modify and add bash script to conform to cross-project STRTA standard.

Closes #161.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -24,7 +24,9 @@ node {
             export const apiHost = \'${env.API_HOST}\';
             export const defaultScenario = \'RCP85\';\n'''
 
-        sh 'scripts/cibuild.sh'
+        sh 'scripts/setup'
```
Comment (tnation14) [inline Jenkinsfile]: The [Jenkinsfile best practices](https://www.cloudbees.com/blog/top-10-best-practices-jenkins-pipeline-plugin) suggest keeping each unit of work in its own`stage` block to visually represent the steps of the build. I think we'd want `setup`, `test`, `cibuild` and `cipublish` blocks in this project.

File: scripts/cipublish [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        cd /vagrant
+
+        if [ -z "${S3_WEBSITE_CONFIG_DIR}" ]; then
+            echo -e '\nEnvironment variable S3_WEBSITE_CONFIG_DIR must be set in order to publish.'
+            echo -e "\nSet to '.' to publish to staging, or to 'production_s3' for production."
+            exit 1
+        fi
+
+        echo -e '\nPublishing to s3...'
+        s3_website push --config-dir=${S3_WEBSITE_CONFIG_DIR}
```
Comment (tnation14) [inline scripts/cipublish]: `shellcheck` is throwing an error for me because `${S3_WEBSITE_CONFIG_DIR}` is not double-quoted.

File: scripts/test [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    else
+        cd /vagrant
+
+        echo -e '\nRunning tests...'
+        npm run test
+
+        echo -e '\nRunning linter...'
+
+        # run linter with console output
+        npm run lint
```
Comment (tnation14) [inline scripts/test]: Can we put this in an `else` block below so that the linter doesn't run twice?

File: scripts/setup [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    "Usage: $(basename "$0")
+    Builld vagrant VM."
+}
+
+if [ "${BASH_SOURCE[0]}" = "${0}" ]; then
+    if [ "${1:-}" = "--help" ]; then
+        usage
+    else
+        # run vagrant from up one directory relative to this script
+        cd "$(dirname "$0")" && cd ..
```
Comment (tnation14) [inline scripts/setup]: Minor nitpick, but I think `cd "$(dirname "$0")/.."` will work here.

File: Jenkinsfile [hunk 5]
```diff
@@ -51,7 +53,7 @@ node {
                           credentialsId: 'CCLAB_AWS_CLOUDFRONT_ID',
                           variable: 'CCLAB_AWS_CLOUDFRONT_ID']]) {
           wrap([$class: 'AnsiColorBuildWrapper']) {
-            sh 'scripts/cipublish.sh'
+            sh 'vagrant ssh -c "/vagrant/scripts/cibuild && /vagrant/scripts/cipublish"'
```
Comment (tnation14) [inline Jenkinsfile]: Same comment as above re: splitting into steps.

File: scripts/test (hunk 3 above)
Comment (flibbertigibbet) [inline scripts/test]: This is run twice intentionally on Jenkins, so the console output is available for reading without having to run it locally.

File: Jenkinsfile (hunk 1 above)
Comment (flibbertigibbet) [inline Jenkinsfile]: Sure.

**Pull request merged**